src/main/java/com/ibm/cloud/cloudant/kafka/transforms/predicates/IsDesignDocument.java

```diff
@@ -45,14 +47,13 @@ public ConfigDef config() {
     public boolean test(SourceRecord record) {
         Schema schema = record.keySchema();
         Object recordKey = record.key();
-        if (schema == null || recordKey == null || schema.type() != Type.MAP || schema.keySchema().type() != Type.STRING || schema.valueSchema().type() != Type.STRING) {
-            // Unexpected schema, check if the key is a map
-            // No need to use message file here, because this purpose gets inserted into the middle of an untranslated Kafka message
-            Requirements.requireMap(recordKey, "record key in IsDesignDocument predicate");
-            // If it is not a map this will throw
-        }
-        Map<String, String> key = (Map<String, String>) recordKey;
-        String id = key != null ? key.get(CLOUDANT_DOC_ID) : null;
+        // No need to use message file here, because this purpose gets inserted into the middle of an untranslated Kafka message
+        Requirements.requireSchema(schema, "record key in IsDesignDocument predicate");
+        Requirements.requireStruct(recordKey, "record key in IsDesignDocument predicate");
+        Struct key = (Struct) recordKey;
+        // will throw DataException if not valid
+        key.validate();
+        String id = key.getString(CLOUDANT_DOC_ID);
         if (id == null) {
```

Comment:
I guess this check is no longer necessary as the key schema has a required ID and would throw earlier during `validate` if the id was `null`?